clang-tools-extra/clang-tidy/tool/ClangTidyMain.cpp

```diff
@@ -553,6 +555,20 @@ static llvm::IntrusiveRefCntPtr<vfs::OverlayFileSystem> createBaseFS() {
 
 int clangTidyMain(int argc, const char **argv) {
   llvm::InitLLVM X(argc, argv);
+  SmallVector<const char *> Args{argv, argv + argc};
+
+  llvm::BumpPtrAllocator Alloc;
```

Comment:
Can we add a one-line comment describing what this block of code does/is meant for? Even better, can it be put into its own function?

Reply:
extract to another function is impossible because variable lifetime issue. The allocator must have same lifetime with argc
 and argv.